### PR TITLE
Allow user to list only pending tasks or all tasks

### LINKED ISSUE #4: Allow users to list only pending tasks or all tasks
As a user if I type `list` I expect to see only pending tasks, but if I type `list --all` I expect to see all tasks (pending and completed).

### PR DESCRIPTION
**Pull request by TomomiKimura**
Resolves #4 

| Case | Screenshot |
| --- | --- |
| `list` | Screen Shot 2022-01-10 at 21 00 06 |
| `list --all` | Screen Shot 2022-01-10 at 21 01 04 |

### CODE REVIEW

File: src/handlers/list.ts [hunk 1]
```diff
@@ -9,24 +9,34 @@ export namespace ListHandler {
...
+    let allTasks: Array<Task>;
     try {
       const data = await readFile(taskData);
       const stringContent = data.toString();
-      taskObjects = JSON.parse(stringContent);
+      allTasks = JSON.parse(stringContent);
     } catch (e: unknown) {
       console.error("Error! We couldn't read the file...");
       return;
     }
```
Comment (MiguelAraCo) [inline src/handlers/list.ts:20]: We may want to move this to a separate function so that it can be reused

File: src/handlers/list.ts [hunk 2]
```diff
@@ -9,24 +9,34 @@ export namespace ListHandler {
...
+    const tasksToPrint: Array<Task> = args.all ? allTasks : pendingTasks;
+
+    // map function : changing Array<string> to Array <string> + [] or [x]
+    const taskStrings: Array<string> = tasksToPrint.map((task) => {
       if (task.done) {
         return `[x] ${task.description}`;
       } else {
         return `[ ] ${task.description}`;
       }
     });
```
Comment (MiguelAraCo) [inline src/handlers/list.ts:36]: Same here

File: src/handlers/list.ts [hunk 3]
```diff
@@ -9,24 +9,34 @@ export namespace ListHandler {
...
-    for (const eachTask of taskStrings) {
-      console.log(eachTask);
+
+    for (const taskString of taskStrings) {
+      console.log(taskString);
     }
```
Comment (MiguelAraCo) [inline src/handlers/list.ts:40]: Same here

**Review by MiguelAraCo — APPROVED** [review]
Added a couple of comments but overall looks good! The changes that I listed don't need to be addressed as part of this PR. I'll leave it up to you if you want to do it now or later.

File: src/handlers/list.ts (hunk 1 above)
Comment (TomomiKimura) [reply src/handlers/list.ts:20]: Yes that sounds better! Thank you for your comment and I will do it as a part of next PR.

File: src/handlers/list.ts (hunk 2 above)
Comment (TomomiKimura) [reply src/handlers/list.ts:36]: I got it!

File: src/handlers/list.ts (hunk 3 above)
Comment (TomomiKimura) [reply src/handlers/list.ts:40]: OK!

**Pull request merged**